File: TheLibraryj/src/main/java/org/library/thelibraryj/book/dto/BookResponse.java
package org.library.thelibraryj.book.dto;

import org.library.thelibraryj.book.domain.BookState;
import org.library.thelibraryj.book.domain.BookTag;

import java.util.List;
import java.util.UUID;

public record BookResponse(UUID id,
                           String title,
                           String author,
                           String description,
                           int chapterCount,
                           float averageRating,
                           int ratingCount,
                           List<ChapterPreviewResponse> chapterPreviews,
                           List<RatingResponse> ratings,
                           List<BookTag> bookTags,
                           BookState bookState,
                           byte[] coverImage) {
}

File: TheLibraryA/src/app/book/book-response.ts
import {ChapterPreview} from './chapter-preview';
import {RatingResponse} from './rating-response';
import {BookState} from './BookState';
import {BookTag} from './BookTag';

export interface BookResponse {
  id : string,
  title: string;
  author: string;
  description: string;
  chapterCount: number;
  averageRating: number;
  ratingCount: number;
  chapterPreviews: ChapterPreview[];
  ratings: RatingResponse[];
  bookTags: BookTag[],
  bookState: BookState;
  coverImage: string;
}
